core/gesture.js

```diff
@@ -760,10 +760,10 @@ Blockly.Gesture.prototype.doBlockClick_ = function() {
  */
 Blockly.Gesture.prototype.doWorkspaceClick_ = function(e) {
   Blockly.keyboardAccessibilityMode_ = false;
-  Blockly.cursor.hide();
+  this.creatorWorkspace_.cursor.hide();
   if (e.shiftKey) {
     // Show the cursor at the specified position
-    Blockly.cursor.workspaceShow(e);
+    this.creatorWorkspace_.cursor.workspaceShow(e);
```

Comment:
It might be cleaner to call cursor.setLocation(this.creatorWorkspace_, false, [x, y]) where x and y are coordinates (possibly as workspace units).

This means you have one code path for moving the cursor. Using workspace units for x and y means you can move the cursor on a headless workspace as well.